### ISSUE TITLE
Question:  sibling dependencies from monorepo will be inlined?

### ISSUE THREAD

**Issue by WaiSiuKei** [timocov/dts-bundle-generator#91]
I am developing a monorepo [https://github.com/WaiSiuKei/fin](https://github.com/WaiSiuKei/fin) with yarn workspace feature and lerna, when building [one package](https://github.com/WaiSiuKei/fin/tree/master/packages/keyboard) which dependents on its [sibling](https://github.com/WaiSiuKei/fin/tree/master/packages/platform), I find the defination of that silbing is inlined. I have tried to add following lines to the [config file](https://github.com/WaiSiuKei/fin/blob/master/packages/keyboard/dts.config.js), but does not works:
```
      libraries: {
        allowedTypesLibraries: ['@fin/platform'],
      // or
        importedLibraries: ['@fin/platform'],
      },
```
Only after adding the path mappings to the [tsconfig.json](https://github.com/WaiSiuKei/fin/blob/master/packages/tsconfig.json#L14-L18), I got the expected output. 
So, it is that path mapping is the only one way to solve the problem, or am I missing something important in the config file for dts-bundle-generator?

**Comment by timocov** [comment]
Hm, interesting... Can you please try to disable `followSymlinks` config option and try again? It might be that lerna and/yarn uses symlinks, which are resolved by the compiler to original one and `dts-bundle-generator` treats that files as local ones.

**Comment by timocov** [comment]
The original problem (that `dts-bundle-generator` doesn't respect import's path you use) is "a little bit" complicated. We even have a label [detect-import-path-problem](https://github.com/timocov/dts-bundle-generator/issues?q=is%3Aissue+is%3Aopen+label%3Adetect-import-path-problem) for such issues, but for now I'm not sure how to fix it properly. I think about this issue time to time, but haven't a solution yet, unfortunately.

**Comment by WaiSiuKei** [comment]
Thank you very much, `followSymlinks: false` works like a charm.

**Comment by paztis** [comment]
I encounter the same kind of problems and followSymlinks didn't change anything
I've a big monolithic repository managed by yarn.
I've got 200 modules developed inside

For one of these modules, Ilve declared the dependency to dts-bundle-generator. But as it is a mono repo, everything is hoisted on the root level of the repo.

When I run dts-bundle-generator, it seems it is trying to integrate everything in my root node_modules
At the begining I've the warning
```
WARNING: It seems that some files in the compilation still are not declaration files.
For more information see https://github.com/timocov/dts-bundle-generator/issues/53.
If you think this is a mistake, feel free to open new issue or just ignore this warning.

  digitalexp-ui-base/node_modules/events/events.js
  digitalexp-ui-base/node_modules/util/util.js
  digitalexp-ui-base/node_modules/url/url.js
  digitalexp-ui-base/node_modules/querystring/index.js
  digitalexp-ui-base/node_modules/@babel/runtime/helpers/interopRequireWildcard.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/get-queries-for-element.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/queries/index.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/query-helpers.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/wait.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/wait-for-element.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/wait-for-element-to-be-removed.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/wait-for-dom-change.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/matches.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/get-node-text.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/events.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/screen.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/role-helpers.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/pretty-dom.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/config.js
  digitalexp-ui-base/node_modules/@testing-library/dom/dist/index.js
  digitalexp-ui-base/node_modules/unist-util-stringify-position/index.js
  digitalexp-ui-base/node_modules/vfile-message/index.js

```

I didn't use any of these modules in my app> They are just present in the root node module because of webpack devDependency.

But after it makes everything crash because of 
```
Error: Cannot find symbol for node: _maxListeners
```
This is a variable inside `events` library that is not at all typescript compliant.

`followSymlinks` to false didn't change anything.
I wonder how you resolve the sources list, because none of these files are at all used in the typescript code I want to bundle.
Is it because of the @types management ?

**Comment by timocov** [comment]
@paztis can you please create small repro for your issue?

> It seems that some files in the compilation still are not declaration files.

means that in the compilation there is files under "node_modules", which are TypeScript files (`.ts`) rather than declaration ones (`.d.ts`) and tsc won't compile them at all. Quite probably that you need to generate declaration files for all projects you want to bundle declarations first (and for all their deps). But I need to be able to repro that to play with that so I can try to help.
